lib/main.py

```diff
@@ -27,19 +29,27 @@ async def lifespan(_app: FastAPI):
 
 
 APP = FastAPI(lifespan=lifespan)
-TASK_LIST: queue.Queue = queue.Queue(maxsize=100)
 
 
 class BackgroundProcessTask(threading.Thread):
     def run(self, *args, **kwargs):  # pylint: disable=unused-argument
+        nc = NextcloudApp()
+        # Hack to workaround auth problems
+        nc.set_user("admin")
+
         while True:
-            task = TASK_LIST.get(block=True)
+            response = nc.providers.task_processing.next_task("core:text2text")
+            if not isinstance(response, dict):
+                continue
```

Comment:
If the response is empty we should have a timeout to avoid constantly spamming the server. If there was a task then we don't need to wait.